### DISCUSSION TITLE
Buildroot Login Fail with Password Error(bad salt)

### DISCUSSION THREAD

**Discussion by chu551122** [polarfire-soc/polarfire-soc-documentation#198] [Linux build systems for PolarFire SoC] [answered]
Dear Sir:
   I git the Buildroot on 9/3/2022 from Polarfire SoC Git and built it to program to Polarfire SoC Icicle Kit.
   When I powered it to Buildroot Log In with root, it asked me the password even I didn't set password in the menuconfig.
   It showed "Login: bad salt" and I didn't understand what happened.
   Can you help me check what's wrong and how to solve this issue?
   p.s. I've done the same procedure successfully previously(I've stored the previous-build image in SD Card and I captured the Login successfully screenshot at the last for your reference)
   Thank you very much!
![image](https://user-images.githubusercontent.com/55179520/188251829-4f703916-312a-4d4b-8c1b-e2e138e0adc8.png)
![image](https://user-images.githubusercontent.com/55179520/188251844-d47c82af-db1c-447e-9770-7fce5602efcb.png)
![image](https://user-images.githubusercontent.com/55179520/188251999-f5b8a31f-6ecb-4746-b721-86af9e2a57a9.png)

**Comment by ConchuOD** [comment, marked as answer]
Hey Michael,
Looking at the Kconfig symbol for that option:
```kconfig
config BR2_TARGET_ENABLE_ROOT_LOGIN
	bool "Enable root login with password"
	default y
	select BR2_PACKAGE_HOST_MKPASSWD if BR2_TARGET_GENERIC_ROOT_PASSWD != ""
	help
	  Allow root to log in with a password.

	  If not enabled, root will not be able to log in with a
	  password. However, if you have an ssh server and you add an
	  ssh key, you can still allow root to log in. Alternatively,
	  you can use sudo to become root.
```

Looks like by un-setting this config option you remove the ability to log in at all!

To get a root login without a password, in the buildroot sdk we use:
BR2_TARGET_ENABLE_ROOT_LOGIN=y
BR2_TARGET_GENERIC_ROOT_PASSWD=""

Hope that helps!
Conor.

**Reply by chu551122** [reply]
Dear Conor:
    Thank you very much for your reply, I found my issue(login : bad salt) might due to I didn't do "unset RISCV" before "make all"(I'm not sure why), after "unset RISCV" then programming to the Icicle Kit, then I can login successfully.
    However, I met another strange issue for password, I captured the ".config", I marked the password "xxxxxx" on purpose, with such ".config" to make the buildroot and program to the Icicle kit, I can log in but no "password" request this time even I've set the password in the menuconfig.
   Can you indicate me which file I need to check besides the ".config" in /work/buildroot_initramfs directory?
   Thank you very much!

![image](https://user-images.githubusercontent.com/55179520/188419120-c507b064-6df5-4479-8b08-90a2535223fe.png)

**Reply by ConchuOD** [reply]
> I need to check besides the ".config" in /work/buildroot_initramfs directory? Thank you very much!

Take a look at system/Config.in for the various password related options:

https://github.com/buildroot/buildroot/blob/master/system/Config.in

**Reply by chu551122** [reply]
I think this is the case I selected at ".config". 
As I mentioned that the password request didn't happened when I log in buildroot with this setting. 
In order to make "scp" file transfer from host PC to Icicle kit. so I used "passwd" command at Polarfire SoC Icicle Kit to enable the "password" alternatively.
FYR
Thank you very much!
![image](https://user-images.githubusercontent.com/55179520/188519858-691cea73-4188-46ad-9b21-3a5b0e4a7394.png)
